test/tests/interfaces/pbs_sched_interface_test.py

```diff
@@ -0,0 +1,143 @@
+        self.assertTrue(rc == 0)
+
+    def test_list_schedulers(self):
+        """
+        List a scheduler object.
+        """
+        pass
+
+    def test_delete_schedulers(self):
+        """
+        Delete a scheduler object succcessfully .
+        """
+        a = {'sched_cycle_length': 12000}
+        rc = self.server.manager(MGR_CMD_CREATE, SCHED, a, id="test1")
+        rc = self.server.manager(MGR_CMD_DELETE, SCHED, a, id="test1")
+        self.assertTrue(rc == 0)
+
+    def test_duplicate_scheduler_name(self):
+        """
+        Check for the scheduler object name.
+        """
+        a = {'sched_cycle_length': 12000}
+        rc = self.server.manager(MGR_CMD_CREATE, SCHED, a, id="test1")
+        try:
+            rc = self.server.manager(MGR_CMD_CREATE, SCHED, a, id="test1")
+        except PbsManagerError, e:
+            self.assertTrue(
+                'qmgr: Error (15211) returned from server' in e.msg[1])
+
+    def test_permission_issue_on_sched(self):
+        """
+        Check for the permission to create/delete/modify scheduler object.
+        """
+        try:
+            self.server.manager(MGR_CMD_CREATE, SCHED,
+                                id="test1",
+                                runas=OPER_USER)
+        except PbsManagerError, e:
+            if self.server.get_op_mode() == PTL_CLI:
+                self.assertTrue(
+                    'qmgr: Error (15007) returned from server' in e.msg[1])
+
+        try:
+            self.server.manager(MGR_CMD_CREATE, SCHED,
+                                id="test2",
+                                runas=ROOT_USER)
+        except PbsManagerError, e:
+            self.fail(e.msg)
+
+    def test_delete_default_sched(self):
+        """
+        Delete default scheduler.
+        """
+        a = {}
+        try:
+            rc = self.server.manager(MGR_CMD_DELETE, SCHED, a, id="main")
+        except PbsManagerError, e:
+            if self.server.get_op_mode() == PTL_CLI:
+                self.assertTrue(
+                    'qmgr: Error (15214) returned from server' in e.msg[1])
+
+    def test_set_sched_attributes(self):
+        """
+        Set scheduler object attribute.
+        """
+        a = {}
+        # Default values will be set
+        rc = self.server.manager(MGR_CMD_CREATE, SCHED, a, id="test1")
+        a = {'sched_cycle_length': 1234}
+        rc = self.server.manager(MGR_CMD_SET, SCHED, a, id="test1")
+        self.assertTrue(rc == 0)
+        # TODO call the list using qmgr and assert the attribute
+
+    def test_unset_sched_attributes(self):
+        """
+        Unset scheduler object attribute.
+        """
+        a = {'sched_cycle_length': 1234}
+        rc = self.server.manager(MGR_CMD_CREATE, SCHED, a, id="test1")
+
+        rc = self.server.manager(MGR_CMD_UNSET, SCHED,
+                                 'sched_cycle_length', id="test1")
+        self.assertTrue(rc == 0)
```

Comment:
This is definitely a case where you want to test the value of sched_cycle_length.  It returns to the default of 1200 after it is unset, so this should be checked.  Speaking of that, you might want to check all of the attributes which return to their default state after they are unset.